src/Utilities/universalfallback.jl

```diff
@@ -173,41 +173,76 @@ end
 # Name
 # The names of constraints not supported by `uf.model` need to be handled
 function MOI.set(uf::UniversalFallback, attr::MOI.ConstraintName, ci::CI{F, S}, name::String) where {F, S}
-    # TODO: Switch to lazily building the name-to-con dict like Model does.
-    if check_can_assign_name(uf, CI, ci, name)
-        if MOI.supports_constraint(uf.model, F, S)
-            MOI.set(uf.model, attr, ci, name)
-        else
-            setname(uf.connames, uf.namescon, ci, name)
-        end
+    if MOI.supports_constraint(uf.model, F, S)
+        MOI.set(uf.model, attr, ci, name)
+    else
+        uf.con_to_name[ci] = name
+        uf.name_to_con = nothing # Invalidate the name map.
     end
+    return
 end
 function MOI.get(uf::UniversalFallback, attr::MOI.ConstraintName, ci::CI{F, S}) where {F, S}
     if MOI.supports_constraint(uf.model, F, S)
         return MOI.get(uf.model, attr, ci)
     else
-        return get(uf.connames, ci, EMPTYSTRING)
+        return get(uf.con_to_name, ci, EMPTYSTRING)
+    end
+end
+
+function build_name_to_con_map(uf::UniversalFallback)
+    uf.name_to_con = Dict{String, CI}()
+    for (con, con_name) in uf.con_to_name
+        if haskey(uf.name_to_con, con_name)
+            # -1 is a special value that means this string does not map to
+            # a unique constraint name.
+            uf.name_to_con[con_name] = CI{Nothing, Nothing}(-1)
+        else
+            uf.name_to_con[con_name] = con
+        end
     end
+    return
 end
 
 MOI.get(uf::UniversalFallback, ::Type{VI}, name::String) = MOI.get(uf.model, VI, name)
 function MOI.get(uf::UniversalFallback, ::Type{CI{F, S}}, name::String) where {F, S}
+    if uf.name_to_con === nothing
+        build_name_to_con_map(uf)
```

Comment:
Why isn't this only called in the `else` clause below in the case the constraint is not supported ?